units/commit_queue.go

```diff
@@ -0,0 +1,352 @@
+	if err != nil {
+		return "", errors.Wrap(err, "can't get GitHub token from Evergreen Config")
+	}
+
+	return githubToken, nil
+}
+
+func validatePR(pr *github.PullRequest) error {
+	catcher := grip.NewSimpleCatcher()
+
+	if pr.GetMergeCommitSHA() == "" {
+		catcher.Add(errors.New("no merge commit SHA"))
+	}
+	if pr.GetUser() == nil || pr.GetUser().GetID() == 0 {
+		catcher.Add(errors.New("no valid user"))
+	}
+	if pr.GetBase() == nil || pr.GetBase().GetSHA() == "" {
+		catcher.Add(errors.New("no valid base SHA"))
+	}
+	if pr.GetBase() == nil || pr.GetBase().GetRepo() == nil || pr.GetBase().GetRepo().GetName() == "" {
+		catcher.Add(errors.New("no valid base repo name"))
+	}
+	if pr.GetBase() == nil || pr.GetBase().GetRepo() == nil || pr.GetBase().GetRepo().GetOwner() == nil || pr.GetBase().GetRepo().GetOwner().GetLogin() == "" {
+		catcher.Add(errors.New("no valid base repo owner login"))
+	}
+	if pr.GetHead() == nil || pr.GetHead().GetSHA() == "" {
+		catcher.Add(errors.New("no valid head SHA"))
+	}
+	if pr.GetNumber() == 0 {
+		catcher.Add(errors.New("no valid pr number"))
+	}
+	if pr.GetTitle() == "" {
+		catcher.Add(errors.New("no valid title"))
+	}
+
+	return catcher.Resolve()
+}
+
+func makeVersion(ctx context.Context, githubToken string, projectRef *model.ProjectRef, pr *github.PullRequest) (*model.Version, error) {
+	config, err := getProjectConfigFromFileInCommit(ctx, githubToken, projectRef, *pr.MergeCommitSHA)
+	if err != nil {
+		return nil, errors.Wrap(err, "can't get remote config file")
+	}
+
+	patchDoc, err := makeMergePatch(pr, projectRef.Identifier, config)
+	if err != nil {
+		return nil, errors.Wrap(err, "can't make patch")
+	}
+
+	if err = patchDoc.Insert(); err != nil {
+		return nil, errors.Wrap(err, "can't insert patch")
+	}
+
+	v, err := model.FinalizePatch(ctx, patchDoc, evergreen.MergeTestRequester, githubToken)
+	if err != nil {
+		return nil, errors.Wrap(err, "can't finalize patch")
+	}
+
+	return v, nil
+}
+
+func getProjectConfigFromFileInCommit(ctx context.Context, githubToken string, projectRef *model.ProjectRef, sha string) (string, error) {
+	cancelCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	owner := projectRef.Owner
+	repo := projectRef.Repo
+	file := projectRef.CommitQConfigFile
+	remoteConfigFile, err := thirdparty.GetGithubFile(cancelCtx, githubToken, owner, repo, file, sha)
+	if err != nil {
+		return "", errors.Wrapf(err, "error fetching file %s from commit %s", file, sha)
+	}
+	configFileContents, err := base64.StdEncoding.DecodeString(*remoteConfigFile.Content)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to decode file %s", file)
+	}
+
+	return string(configFileContents), nil
+}
+
+func makeMergePatch(pr *github.PullRequest, projectID string, configString string) (*patch.Patch, error) {
+	u, err := getPatchUser(pr.GetUser().GetID())
+	if err != nil {
+		return nil, errors.Wrap(err, "can't get user for patch")
+	}
+	prAuthor := u.Id
```

Comment:
throw away variables... I don't know that this makes things more clear, really.

Reply:
removed